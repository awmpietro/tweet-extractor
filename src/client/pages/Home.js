import React, { useState } from 'react';
import {
  Container,
  TabContent,
  TabPane,
  Nav,
  NavItem,
  NavLink,
  Card,
  Row,
  Col,
  Alert,
} from 'reactstrap';
import classnames from 'classnames';
import axios from 'axios';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch } from '@fortawesome/free-solid-svg-icons';

import Spin from '../components/Spin/Spin';
import RenderTweets from '../components/RenderTweets';
import LanguageStats from '../components/LanguageStats';
import HashtagStats from '../components/HashtagStats';
import UserLocationStats from '../components/UserLocationStats';
import DateStats from '../components/DateStats';
import GeneralStats from '../components/GeneralStats';
import CsvFormatter from '../components/CsvFormatter';

import useInput from '../hooks/useInput';

import '../assets/style.css';

export default function Home() {
  const [activeTab, setActiveTab] = useState('1');
  const [loading, setLoading] = useState(false);
  const [results, setResults] = useState([]);
  const [stats, setStats] = useState({});
  const [isError, setIsError] = useState(false);
  const [errorMsg, setErrorMsg] = useState('');

  const { value, bind, reset } = useInput('');

  const toggleTab = (tab) => {
    if (activeTab !== tab) setActiveTab(tab);
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      setLoading(true);
      const res = await axios.post(process.env.REACT_APP_API_URL, {
        data: value,
      });
      setResults(res.data.data);
      setStats(res.data.stats);
      reset();
      setLoading(false);
    } catch (error) {
      setErrorMsg(error.response.data);
      setIsError(true);
      setLoading(false);
    }
  };

  if (!loading) {
    return (
      <Container style={{ paddingTop: 50 }}>
        <Row>
          <Col sm="12">
            <h1>
              Tweet <small> Extractor</small>
            </h1>
          </Col>
        </Row>
        <hr />
        <Nav tabs>
          <NavItem>
            <NavLink
              className={classnames({ active: activeTab === '1' })}
              onClick={() => {
                toggleTab('1');
              }}
            >
              Search
            </NavLink>
          </NavItem>
          {results.length > 0 ? (
            <NavItem>
              <NavLink
                className={classnames({ active: activeTab === '2' })}
                onClick={() => {
                  toggleTab('2');
                }}
              >
                Statistics
              </NavLink>
            </NavItem>
          ) : null}
        </Nav>
        <TabContent activeTab={activeTab}>
          <TabPane tabId="1">
            <Row className="mt-3">
              <Col sm="12">
                <form onSubmit={handleSubmit}>
                  <Row>
                    <Col sm="10">
                      <div className="form-group">
                        <input
                          className="form-control form-control-lg"
                          type="text"
                          placeholder="Enter search terms comma separated"
                          {...bind}
                          required
                        />
                      </div>
                    </Col>
                    <Col sm="2" className="text-right">
                      <div className="form-group">
                        <button
                          type="submit"
                          className="btn btn-lg btn-primary"
                          id="search-btn"
                        >
                          <FontAwesomeIcon color="#fff" icon={faSearch} />{' '}
                          Search
                        </button>
                      </div>
                    </Col>
                  </Row>
                </form>
              </Col>
            </Row>
            {results.length > 0 ? (
              <Row>
                <Col sm="12">
                  <CsvFormatter results={results} />
                </Col>
                <Col sm="12">
                  <hr />
                </Col>
              </Row>
            ) : null}
            <RenderTweets results={results} />
          </TabPane>
          {results.length > 0 ? (
            <TabPane tabId="2">
              <Row className="mt-3">
                <Col md={12} lg={12} xs={12}>
                  <Card>
                    <Card body>
                      <DateStats stats={stats} />
                      <hr />
                      <LanguageStats stats={stats} />
                      <hr />
                      <HashtagStats stats={stats} />
                      <hr />
                      <UserLocationStats stats={stats} />
                      <hr />
                      <GeneralStats stats={stats} />
                    </Card>
                  </Card>
                </Col>
              </Row>
            </TabPane>
          ) : null}
        </TabContent>
        <Alert color="danger" isOpen={isError} toggle={() => setIsError(false)}>
          {errorMsg}
        </Alert>
      </Container>
    );
  }
  return <Spin />;
}
