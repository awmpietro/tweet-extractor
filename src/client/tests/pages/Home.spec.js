import React from 'react';
import Home from '../../pages/Home';
import renderer from 'react-test-renderer';

describe('Home', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Home />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});