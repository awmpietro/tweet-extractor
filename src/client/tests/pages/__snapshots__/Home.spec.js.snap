// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot renders 1`] = `
<div
  className="container"
  style={
    Object {
      "paddingTop": 50,
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col-sm-12"
    >
      <h1>
        Tweet 
        <small>
           Extractor
        </small>
      </h1>
    </div>
  </div>
  <hr />
  <ul
    className="nav nav-tabs"
  >
    <li
      className="nav-item"
    >
      <a
        className="active nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </li>
  </ul>
  <div
    className="tab-content"
  >
    <div
      className="tab-pane active"
    >
      <div
        className="mt-3 row"
      >
        <div
          className="col-sm-12"
        >
          <form
            onSubmit={[Function]}
          >
            <div
              className="row"
            >
              <div
                className="col-sm-10"
              >
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    onChange={[Function]}
                    placeholder="Enter search terms comma separated"
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="text-right col-sm-2"
              >
                <div
                  className="form-group"
                >
                  <button
                    className="btn btn-lg btn-primary"
                    type="submit"
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-search fa-w-16 "
                      color="#fff"
                      data-icon="search"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                     Search
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
